File: benchmark/dmrg.jl
using ITensors
using Printf
#using Logging
##Set global logging level to Debug to show timers
#global_logger(SimpleLogger(stdout, Logging.Debug))

function main()
  N = 100
  sites = spinOneSites(N)

  ampo = AutoMPO(sites)
  for j=1:N-1
    add!(ampo,"Sz",j,"Sz",j+1)
    add!(ampo,0.5,"S+",j,"S-",j+1)
    add!(ampo,0.5,"S-",j,"S+",j+1)
  end
  H = toMPO(ampo)

  psi = randomMPS(sites)

  sweeps = Sweeps(1)
  maxdim!(sweeps,10,20,100,100,200)
  cutoff!(sweeps,1E-11)
  @show sweeps

  energy,psi = @time dmrg(H,psi,sweeps,maxiter=2)
  @printf "Final energy = %.12f\n" energy
end
main()

File: benchmark/dmrg.cc
#include "itensor/all.h"
using namespace itensor;

int 
main()
    {
    int N = 100;

    auto sites = SpinOne(N,{"ConserveQNs=",false});

    auto ampo = AutoMPO(sites);
    for(auto j : range1(N-1))
        {
        ampo += 0.5,"S+",j,"S-",j+1;
        ampo += 0.5,"S-",j,"S+",j+1;
        ampo +=     "Sz",j,"Sz",j+1;
        }
    auto H = toMPO(ampo);

    auto state = InitState(sites);
    for(auto i : range1(N))
        {
        if(i%2 == 1) state.set(i,"Up");
        else         state.set(i,"Dn");
        }
    auto psi0 = MPS(state);

    auto sweeps = Sweeps(5);
    sweeps.maxdim() = 10,20,100,100,200;
    sweeps.cutoff() = 1E-11;
    sweeps.niter() = 2;
    sweeps.noise() = 1E-7,1E-8,0.0;
    println(sweeps);

    auto [energy,psi] = dmrg(H,psi0,sweeps,"Quiet");

    PrintData(inner(psi,H,psi));

    return 0;
    }
